fix(console): fix ENS tests not working with embark side by side

jrainville opened the pull request:
I'd like @emizzle to confirm I didn't do something wrong, but long story short, when the console is a client in terms of IPC, it didn't do the EmbarkJS `connectConsole` so ENS tests were bound to fail. 

I think we could actually remove the `isEmbarkConsole` condition altogether for the register, but maybe it slows things down a bit or we make it so that we eval only through the console? Because there's a weird gap between runCode and console, but they still need one another

emizzle commented on packages/embark/src/lib/modules/console/index.ts:
```diff
@@ -84,6 +84,10 @@ class Console {
   }
 
   private get isEmbarkConsole() {
+    if (this.embark.currentContext && this.embark.currentContext.includes("test")) {
```
Actually, it might be better if we explicitly check that we are running in the context of the console:
```
  private get isEmbarkConsole() {
    return this.ipc.connected && this.ipc.isClient() && this.embark.currentContext.includes(constants.contexts.console);
  }
```

jrainville replied:
That makes a lot of sense

jrainville commented:
@emizzle Changed according to your suggestion.

I also fixed the console and the logger in this commit https://github.com/embark-framework/embark/pull/1364/commits/68ea85f9bb3183ffb733f847afd9fb60f61cf36c

emizzle commented on packages/embark/src/cmd/dashboard/repl.js:
```diff
@@ -19,9 +19,10 @@ class REPL {
-  enhancedEval(cmd, _context, _filename, _callback) {
+  enhancedEval(cmd, _context, _filename, callback) {
     this.events.request('console:executeCmd', cmd.trim(), (err, message) => {
       this.logger.info(message === undefined ? '' : message);
```
Why must we print the message via logger, as opposed to passing the object/message to the `REPL` and letting it handle it's display (as it was before). I understand that we want to print the message in the non-console dashboard, but couldn't that be handled by only logging when detecting an IPC connection (ie `ipc.isServer() && ipc.connected`).

jrainville replied:
It's just that with that way, the logger is the only source of logging. Which handles transmitting to cockpit etc

jrainville commented:
Good catch on the circular error. I forgot about that. It's an easy fix, we already have a function fix those. Gonna change it.

jrainville commented:
@emizzle Fixed by using node's `util.inspect`. It is most likely what REPL itself uses.

emizzle commented:
Found a few errors while testing:

1. The downside of not letting `REPL` handle the object display is we lose some color formatting. For example, before when typing `web3`, I had a nice colored display of the `web3` object. Now it is just one color, due to using the logger. But not a deal breaker I guess. This is the same for displaying arrays, etc.
2. Errors are not shown, ie typing `web.eth` doesn't show anything.
3. Typing `swarm` in the cockpit, shows a 500 error (underlying error displayed in the console is `TypeError: Cannot read property 'green' of undefined`), then displays the correct message: `Swarm is disabled or not configured. Please see https://embark.status.im/docs/storage_configuration.html for more information.`. However, typing `swarm` in to the console shows nothing. This is probably related to errors not showing (the point above).
4. In the console, typing `await EmbarkJS.Names.isAvailable()` shows `undefined` as the result, but if I debug this, `true` is being returned from the VM.

jrainville commented:
@emizzle you will find here https://github.com/embark-framework/embark/pull/1364/commits/101bc43d466e4d742bb1902a9986c5af2094b770 that I reverted the changes to the repl, so the console is back to the old behaviour

What I did keep is that commands put in the cockpit are still outputted in Embark

emizzle commented:
Thanks @jrainville, it's much better now :)

There is still one error that seems like it could be related to the changes:

Typing swarm in the cockpit, shows a 500 error (underlying error displayed in the console is `TypeError: Cannot read property 'green' of undefined`), then continues on to display the correct message: `Swarm is disabled or not configured. Please see https://embark.status.im/docs/storage_configuration.html for more information.`.

Full stack shown in the console:
```
TypeError: Cannot read property 'green' of undefined
    at _logFunction.args.map.arg (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/src/lib/core/logger.js:22:77)
    at Array.map (<anonymous>)
    at Logger.map [as logFunction] (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/src/lib/core/logger.js:20:33)
    at Logger.info (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/src/lib/core/logger.js:125:23)
    at info (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/src/lib/modules/console/index.ts:107:23)
    at Console.cb (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/src/lib/modules/swarm/index.js:36:11)
    at Console.call [as executeCmd] (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/src/lib/modules/console/index.ts:174:33)
    at executeCmd (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/src/lib/modules/console/index.ts:100:12)
    at cb (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/src/lib/modules/api/server.ts:133:7)
    at EventEmitter.cb (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/src/lib/modules/authenticator/index.js:123:32)
    at EventEmitter.call (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/src/lib/core/events.js:75:8)
    at EventEmitter.emit (events.js:182:13)
    at EventEmitter.emit (domain.js:442:20)
    at EventEmitter.emit [as request] (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/src/lib/core/events.js:69:15)
    at Server.request [as applyHTTPFunction] (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/src/lib/modules/api/server.ts:129:24)
    at Layer.handle [as handle_request] (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/node_modules/express/lib/router/layer.js:95:5)
    at next (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/node_modules/express/lib/router/route.js:137:13)
    at Route.dispatch (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/node_modules/express/lib/router/route.js:112:3)
    at Layer.handle [as handle_request] (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/node_modules/express/lib/router/layer.js:95:5)
    at /Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/node_modules/express/lib/router/index.js:281:22
    at Function.process_params (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/node_modules/express/lib/router/index.js:335:12)
    at next (/Users/emizzle/Code/__Github/embk-fw/embark/packages/embark/node_modules/express/lib/router/index.js:275:10)
```

jrainville commented:
@emizzle Thanks again for all your tests.
I fixed the undefined error.
I also tested with objects again in the cockpit, and since we limited html injections in the cockpit, printing objects from the console to the cockpit didn't work (it was showing html entities), so I reverted that too.
